buffer/src/test/java/io/netty/buffer/UnsafeByteBufUtilTest.java

```diff
@@ -45,4 +47,38 @@ public void testSetBytesOnReadOnlyByteBuffer() throws Exception {
             targetBuffer.release();
         }
     }
+
+    @Test
+    public void testSetBytesOnReadOnlyByteBufferWithPooledAlloc() throws Exception {
+        byte[] testData = new byte[]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+        int length = testData.length;
+
+        ByteBuffer readOnlyBuffer = ByteBuffer.wrap(testData).asReadOnlyBuffer();
+
+        int pageSize = 4096;
+
+        // create memory pool with one page
+        ByteBufAllocator alloc = new PooledByteBufAllocator(true, 1, 1, pageSize, 0);
+        UnpooledDirectByteBuf targetBuffer = new UnpooledDirectByteBuf(alloc, length, length);
+
+        ByteBuf b1 = alloc.heapBuffer(16);
+        ByteBuf b2 = alloc.heapBuffer(16);
```

Comment:
can you also please call release on b1 and b2 just for the sake of consistency :)